### PR TITLE
[WIP] Use alt for render

### PR DESCRIPTION
**Pull request by iam4x**
See #68 

(Still having issues)

### CODE REVIEW

**Comment by transedward** [comment]
why drop alt instances to singleton, any pros and cons?
seems like AltIso.render doesn't conflict on that? any other consideration?

**Comment by transedward** [comment]
hope to see you complete the change and full example
@iam4x ++

**Comment by iam4x** [comment]
@transedward I have issues with `alt` and `react-router` at page change.

And there's a problem with resolving promises at rendering, only the top component promises will be resolved. With `alt-resolver` from the boilerplate you can resolve every promises thrown from components.

**Comment by transedward** [comment]
> only the top component promises will be resolved

seems like a problem with other library too.
any issue reported yet?
and what's the issue about react router?

**Comment by iam4x** [comment]
@transedward I didn't have time to trace the error, I'm working on it :)

**Comment by transedward** [comment]
thanks for the hard work, @iam4x

**Comment by goatslacker** [comment]
Will check this out.

**Comment by iam4x** [comment]
@goatslacker I'm going to try with the new `react-router` alpha

**Comment by quicksnap** [comment]
Just came across this repo--really nice stuff, and excited to see progress on this and #68!

**Comment by iam4x** [comment]
@quicksnap Thank's, I replied to you about the current state of next changes in the issue ;)

**Comment by filod** [comment]
LGTM!  thx a lot, hope this can merge to master soon

**Comment by erkieh** [comment]
BTW, A change to rendering in Alt is in progress: https://github.com/goatslacker/alt/pull/364

**Comment by iam4x** [comment]
@erkieh Yes I'm waiting for this PR, it will solve some of the issues I'm having right now.

**Comment by goatslacker** [comment]
I think it solves the react router issues you were having

**Comment by iam4x** [comment]
Yes thank's @goatslacker can't wait for this to be merged, I'm grateful all of the work you put into Alt :clap:

**Comment by goatslacker** [comment]
<3 sorry it's taking so long. This is quite complex and I want to get it right.

**Comment by iam4x** [comment]
I've rebased with my changes on master with the new beta of react-router, still stuck :)

I'll wait for the new PR to be merged until furthers changes.

**Comment by iam4x** [comment]
@goatslacker Issue with `react-router` which don't render component on second transition is fixed now with the `react-router@1.0-beta3`

Waiting for the PR for having nested components promises to be resolved too. :tada:

**Comment by goatslacker** [comment]
YUS...I haven't had time to check for myself, thanks for checking.

I'll put up a PR today outlining some progress.

File: app/utils/promisify.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export default (fn) => new Promise((resolve) => fn((result) => resolve(result)));
```
Comment (dmitry) [inline app/utils/promisify.js]: Is it used yet anywhere?

Comment (iam4x) [inline app/utils/promisify.js]: You are right it's a left over of the rebase, going to remove it. Thanks.

**Comment by goatslacker** [comment]
Btw this is the latest heap of crappy code I wrote to try qnd solve this problem:

https://github.com/goatslacker/alt/pull/393

**Comment by iam4x** [comment]
See #69

**Pull request closed without merging**